packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/impactanalysis/services/InvokeTreeImpactAnalysisToolEventHandler.java

```diff
@@ -106,7 +108,8 @@ public void handle(One<IPayload> payloadSink, Many<ChangeDescription> changeDesc
                 this.editingContextSnapshotService.restoreSnapshot(siriusWebEditingContext, editingContextSnapshot.get());
 
                 if (entryExecutionResult instanceof Success success) {
-                    payload = new InvokeImpactAnalysisSuccessPayload(invokeTreeImpactAnalysisInput.id(), new ImpactAnalysisReport(diff.getObjectsToAttach().size(), diff.getObjectChanges().size(), diff.getObjectsToDetach().size(), List.of()), success.getMessages());
+                    List<String> additionalMessages = (List<String>) success.getParameters().getOrDefault(IMPACT_ANALYSIS_MESSAGES_PARAMETRER_KEY, List.of());
```

Comment:
This provides a way for executors to return additional messages to put in the impact analysis report.
This was not needed for the update of a library, because the impact analysis is dedicated to this feature. Here we have a generic impact analysis for any context menu entry, so we want to be able to pass more information without explicitly mentioning proxy removal information (which is specific to the removal of a dependency).
This is maybe not clean enough for Sirius Web, in this case let me know and I'll rework it.